test/release/examples/primers/arrays.chpl

```diff
@@ -61,176 +72,229 @@ writeln("After assigning its interior values, A is: ", A);
+// * allowing multiple arrays to share a single domain
+//
+// * associating a logical name with an index set
+//
+// * amortizing overheads when storing multiple distributed arrays with the same indices
+//
+// * enabling compiler optimizations.
+//
 //
 // The following domain declaration defines a 2D rectangular domain
-// called ``ProbSpace`` which is the same size and shape as ``B`` was above.
+// called ``ProbSpace``, which has the same size, shape, and index set
+// as ``B`` above.
 //
 
-var ProbSpace: domain(2) = {1..n, 1..n};
+const ProbSpace = {1..n, 1..n};
 
+// Note that we declare the domain as being ``const``, indicating that
+// we will never change the set of indices it represents.  Besides
+// indicating the programmer's intent, this can enable key compiler
+// optimizations, so is recommended whenever a domain's index set is
+// known to be invariant.
 //
-// We then use that domain to declare some arrays...
+
+//
+// We can then use that domain to declare some arrays...
 //
 
 var C, D, E: [ProbSpace] bool;
 
-// ...and to iterate over their index spaces...
+// ...and to iterate over their shared index set...
+
+forall (i,j) in ProbSpace do
+  C[i,j] = (i+j) % 3 == 0;
+
+writeln("After assigning C, its value is:\n", C, "\n");
+
+//
+// An array need not be accessed using indices from the domain that
+// was used to declare it.  For example, the following loop indexes
+// into ``B`` using indices from ``ProbSpace`` even though there is no
+// direct relationship between ``B`` and ``ProbSpace``.
+//
 
 for (i,j) in ProbSpace do
-  C(i,j) = (i+j) % 3 == 0;
+  B[i,j] = i + j/10.0;
 
-writeln("After initializing C, its value is:\n", C, "\n");
+writeln("B has been re-assigned to:\n", B, "\n");
 
 //
-// When indexing over a multidimensional domain, the indices
-// can be captured into a single tuple variable rather than
-// destructuring the tuple into its integer components.
-// Similarly, multidimensional array accesses can be expressed using
-// tuple indices rather than multiple integer arguments.  In the
-// following example, the index variable ij stores a 2-tuple of
-// integers (``2*int`` in Chapel).  This is a really inefficient way
-// to assign the diagonal values "true" -- note the use of tuple
-// indexing to tease the individual components out of the 2-tuple.
+// When iterating over a multidimensional domain, the indices can be
+// expressed using a single tuple variable rather than destructuring
+// the tuple into its integer components.  Similarly, multidimensional
+// array accesses can be written using tuple indices rather than
+// multiple integer arguments.  In the following example, the index
+// variable ``ij`` stores a 2-tuple of integers (``2*int`` or ``(int,
+// int)``).  Note the use of tuple indexing to access the individual
+// components of the 2-tuple ``ij``.
 //
 
 for ij in ProbSpace do
-  D(ij) = ij(0) == ij(1);
-
-writeln("After initializing D, its value is:\n", D, "\n");
+  D[ij] = ij(0) == ij(1);
```

Comment:
Something like:
```chapel
D[ij] = (ij(0) == ij(1)) && C[ij]
```
might help drive the shared-index-space point home.

Reply:
I was thinking that using a loop expression involving more than one of C, D, and E might help to show that they are all defined over the same domain.

> (It has the downside of making only the central element of D true I think?)

That's true, I wasn't thinking about the values themselves. Maybe another expression would be better. It's also definitely okay how it was.

Reply:
I'm going to leave it as-is due to being short on time going into the release.  In retrospect, it does seem a little unfortunate that all of the loops in the primer are currently using a single array only, but that also feels like a topic that's potentially more for the loops primer than the arrays primer (in fact, my draft of the loops primer already has loops that involve multiple arrays).

Thanks!